zcash_client_backend/src/proto/proposal.rs

```diff
@@ -0,0 +1,137 @@
+    /// be executed.
+    #[prost(uint32, tag = "7")]
+    pub min_target_height: u32,
+    /// A flag indicating whether the proposal is for a shielding transaction,
+    /// used for determining which OVK to select for wallet-internal outputs.
+    #[prost(bool, tag = "8")]
+    pub is_shielding: bool,
+}
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct SaplingInputs {
+    /// Returns the Sapling anchor height to be used in creating the transaction.
+    #[prost(uint32, tag = "1")]
+    pub anchor_height: u32,
+    /// The unique identifier and amount for each proposed Sapling input
+    #[prost(message, repeated, tag = "2")]
+    pub inputs: ::prost::alloc::vec::Vec<ProposedInput>,
+}
+/// The unique identifier and amount for each proposed input.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct ProposedInput {
+    #[prost(bytes = "vec", tag = "1")]
+    pub txid: ::prost::alloc::vec::Vec<u8>,
+    #[prost(uint32, tag = "2")]
+    pub index: u32,
+    #[prost(uint64, tag = "3")]
+    pub value: u64,
+}
+/// The proposed change outputs and fee amount.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct TransactionBalance {
+    #[prost(message, repeated, tag = "1")]
+    pub proposed_change: ::prost::alloc::vec::Vec<ChangeValue>,
+    #[prost(uint64, tag = "2")]
+    pub fee_required: u64,
+}
+/// An enumeration of change value types.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct ChangeValue {
+    #[prost(oneof = "change_value::Value", tags = "1")]
+    pub value: ::core::option::Option<change_value::Value>,
+}
+/// Nested message and enum types in `ChangeValue`.
+pub mod change_value {
+    #[allow(clippy::derive_partial_eq_without_eq)]
+    #[derive(Clone, PartialEq, ::prost::Oneof)]
+    pub enum Value {
+        #[prost(message, tag = "1")]
+        SaplingValue(super::SaplingChange),
+    }
+}
+/// An object wrapper for memo bytes, to facilitate representing the
+/// `change_memo == None` case.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct MemoBytes {
+    #[prost(bytes = "vec", tag = "1")]
+    pub value: ::prost::alloc::vec::Vec<u8>,
+}
+/// The amount and memo for a proposed Sapling change output.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct SaplingChange {
+    #[prost(uint64, tag = "1")]
+    pub amount: u64,
+    #[prost(message, optional, tag = "2")]
+    pub memo: ::core::option::Option<MemoBytes>,
+}
+/// The fee rule used in constructing a Proposal
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, ::prost::Enumeration)]
+#[repr(i32)]
+pub enum FeeRule {
+    /// Protobuf requires that enums have a zero descriminant as the default
```

Comment:
Note to self: check this is regenerated when the comment typo (should be "discriminant") is fixed.